xml/Makefile.am

```diff
@@ -145,6 +145,15 @@ api/api-result-%.rng: $(API_files) Makefile.am
 	echo '      </optional>' >> $@
 	echo '      <element name="status">' >> $@
 	echo '        <attribute name="code"> <data type="integer" /> </attribute>' >> $@
+	echo '        <optional>' >> $@
+	echo '          <errors>' >> $@
+	echo '            <oneOrMore>' >> $@
+	echo '              <element name="error">' >> $@
+	echo '                <text />' >> $@
+	echo '              </element>' >> $@
+	echo '            </oneOrMore>' >> $@
+	echo '          </errors>' >> $@
+	echo '        </optional>' >> $@
```

Comment:
I'm not really satisfied with this block, but I think the rest of it is looking pretty good.

Reply:
It's more the fact that there's already a text blob in the status element, and then adding a list of errors in the same spot.  It just seems like duplication.

Reply:
Right, but that text is a simple equivalent of the rc, without any context. The errors can provide context. Perhaps that would be clearer if the rc text were just an attribute of status or in its own element.